ssl/src/main/java/io/crate/protocols/ssl/SslContextProvider.java

```diff
@@ -43,13 +43,14 @@
     private static final String SSL_CONTEXT_METHOD_NAME = "buildSslContext";
 
     private final Settings settings;
-    private final PipelineRegistry pipelineRegistry;
     private SslContext sslContext;
 
     @Inject
     public SslContextProvider(Settings settings, PipelineRegistry pipelineRegistry) {
         this.settings = settings;
-        this.pipelineRegistry = pipelineRegistry;
+        if (SslConfigSettings.isHttpsEnabled(settings)) {
+            pipelineRegistry.registerSslContext(this);
```

Comment:
I think we should always register the context. The PipelineRegistry should decide if it needs the context or not (depending on the setting).